x-pack/auditbeat/module/system/socket/guess/inetsock6.go

```diff
@@ -191,7 +192,7 @@ func (g *guessInetSockIPv6) Probes() (probes []helper.ProbeDef, err error) {
 	raddrOffsets := g.offsets
 	g.offsets = nil
 	const sizePtr = int(sizeOfPtr)
-	var fetch []string
+	fetch := make([]string, 0)
```

Comment:
Why this change? (and the others like it)

Reply:
It's entirely the linter; it complains if you do something like `var list []string`, since it wants you to pre-allocate the array based on size, but for most of these, you don't actually know the size of the array in advance, so `make()` just pre-allocates it to zero.

Reply:
This is one of the multitude of cases where the linter pushes people into writing worse code. IMO we should push back against this. To demonstrate why, this is the comparison between the generated code from the two approaches: https://godbolt.org/z/1zs7fY5f7.

Reply:
I clicked that link, and the `make` version was being compiled as `x86 gc 1.20` versus `x86-64 gc 1.22.1`. After switching the `make` version to be the same, I see this diff:
<img width="1767" alt="Screenshot 2024-08-14 at 7 24 22 AM" src="https://github.com/user-attachments/assets/2c3a867e-3d58-4801-88a5-ea9e64f0d225">

They seem pretty close to me. Am I missing something? (I probably am)

Reply:
This is pretty interesting.  They changed something in 1.21 (released Aug 2023).  If you compare 1.20 or earlier you see that the lint preferred version generates much more code.  Starting with 1.21 they appear to be equivalent.

Reply:
Nice that the compiler is able to optimise this to the same bytes. @nicholasberlin, no you weren't missing anything, @nfritts has summarised perfectly. I guess despite [this](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices) the gc team has made an effort to recognise and re-write the non-idiomatic phrasing.